modules/gdnative/godot/godot_image.cpp

```diff
@@ -105,6 +105,46 @@ int GDAPI godot_image_get_mipmap_count(const godot_image *p_img) {
 	return img->get_mipmap_count();
 }
 
+void GDAPI godot_image_blit_rect(godot_image *p_img, const godot_image *src, const godot_rect2 *src_rect, const godot_vector2 *dest) {
+	Image *img = (Image *)p_img;
+	Image *src_img = (Image *)src;
+	Rect2 *cpp_src_rect = (Rect2 *)src_rect;
+	Vector2 *cpp_dest = (Vector2 *)dest;
+	img->blit_rect(*src_img, *cpp_src_rect, *cpp_dest);
+}
+
+godot_image GDAPI godot_image_compressed(godot_image *p_img, int p_mode) {
+	Image *img = (Image *)p_img;
+	Image cpp_img = img->compressed(p_mode);
+	godot_image *compressed_img = (godot_image *)&cpp_img;
```

Comment:
careful with this, it doesn't work. The destructor will be called at the end of the scope, taking a pointer doesn't change that.

So you need to create an object with non local lifetime using `memnew_placement` and then assign this new image to it.

This simple "pointer casting" technique doesn't work with objects that have a destructor.